File: app/models/event.rb
class Event < ActiveRecord::Base
  attr_accessible :title, :creator_id, :deadline, :final_place, :start_time,
    :end_time, :current_event
  validates :title, :creator_id, :presence => true
  has_many :event_places, :dependent => :destroy
  has_many :places, :through => :event_places, :source => :place
  has_many :time_suggestions, :dependent => :destroy
  has_many :event_circles, :dependent => :destroy
  has_many :members, :through => :event_circles, :source => :user
  has_many :event_pics, :dependent => :destroy
  accepts_nested_attributes_for :event_places, :time_suggestions, :event_circles, allow_destroy: true

  def is_past_event
    @et = self.end_time.strftime('%s')
    @now = Time.zone.now.strftime('%s')
    if @now > @et
      return true
    end
    false
  end
end

File: app/assets/javascripts/models/event.js
PlanIt.Models.Event = Backbone.Model.extend({
  parse: function(response) {
    return {
      title: response.title,
      deadline: response.deadline,
      id: response.id,
      creator_id: response.creator_id,
      created_at: response.created_at,
      updated_at: response.updated_at,
      final_place: response.final_place,
      start_time: response.start_time,
      end_time: response.end_time,
      current_event: response.current_event,
      event_circles: new PlanIt.Collections.EventCircles(response.event_circles),
      event_places: new PlanIt.Collections.EventPlaces(response.event_places),
      time_suggestions: new PlanIt.Collections.TimeSuggestions(response.time_suggestions),
      event_pics: new PlanIt.Collections.EventPics(response.event_pics)
    }
  }
});
